resolver_test.go

```diff
@@ -41,6 +41,12 @@ func TestDeadlineExceeded(t *testing.T) {
 	st.Expect(t, err, context.DeadlineExceeded)
 }
 
+func TestFluentConstructorDeadlineExceeded(t *testing.T) {
```

Comment:
Nice. Would love to see more tests of the fluent constructor options.

Reply:
Added a few, including one for setting the net.Dialer to 127.0.0.1:65353 and that failed in an interesting way (return `RR{}, nil`)